# 每日一问：第一天

### 任务

- 了解数据类型、运算符类型有哪些，不同的数据类型搭配不同的运算符会产生怎样的输出结果，如何进行数据类型转换，字符串的操作方法等等。
- 练习使用jupyter notebook基本命令
    - 切换到命令模式：**Esc**
    - 在区域前插入Cell: **A** 
    - 在区域后插入Cell: **B**
    - 切换成Markdown模式： **M**
    - Markdown双击开始编辑
    - Code和Markdown区域运行：**Shift + Enter**
    - 更多用法你可以参考课程内容，也可以查阅 jupyter notebook [帮助文档](https://jupyter.org/documentation)
- 在练习的过程中，如果你遇到问题，你可以回顾课程内容，也可以查阅 [Python官方文档](https://docs.python.org/3/)
- 请每位同学选择下面与自己学号对应的问题练习， 学有余力的同学也可以完成所有练习


### 问题1：
① x=4，y=5，print（x+y）       ② x='4',y='5', print(x+y)            这两段代码输出的结果相同么？请解释原因。   

In [1]:
# Try your code here
x = 4
y = 5
print(x+y)

x = '4'
y = '5'
print(x+y)


9
45


### 回答1：（双击在下面回答问题）

不相同，①的结果是9，是整型，②的结果是“45”，是字符串。


### 问题2：
‘*’和‘**’运算符执行的操作相同么？请举例说明。

In [2]:
# Try your code here
print(2*3)
print(2**3)

6
8


### 回答2：（双击在下面回答问题）
不同，‘*’运算符表示乘法，如2*3=6，‘**’是幂运算，如2**3=8.

### 问题3：
请写出下式的计算结果：100%(2+2)**3//5

In [3]:
# Try your code here 
print(100%(2+2)**3//5)

7


### 回答3：（双击在下面回答问题）
- 2+2 = 4
- 4**3 = 64
- 100%64 = 36
- 36//5 =7


### 问题4：
'‘10/3'，‘10//3’和'10%3'结果相同么？请说明原因。

In [4]:
# Try your code here 
print(10/3)
print(10//3)
print(10%3)

3.3333333333333335
3
1


### 回答4：（双击在下面回答问题）
不相同，10/3=3.33……，包含小数位,是一般的除法操作；10//3=3，结果向下取整，是整除操作；10%3=1，结果是10除以3的余数，是取余操作。

### 问题5：
浮点型和整型可以调用islower么？执行"Udacity".islower()和"udacity".islower()分别会得到什么结果？为什么？

In [5]:
# Try your code here 
print("Udacity".islower())
print("udacity".islower())


False
True


### 回答5：（双击在下面回答问题）
浮点型和整型不可以调用islower，islower是判断字符串中是否都是小写字母，所以"Udacity".islower()得到的结果为False，"udacity".islower()得到结果为True。

### 问题6：
什么是科学计数法？51200000.0如何用python语法中的科学计数法表示？


In [6]:
# Try your code here 
print(5.12e7)

51200000.0


### 回答6：（双击在下面回答问题）
科学记数法是一种记数的方法。把一个数表示成a（1≤a<10，n为整数）与10的幂相乘的形式，这种记数法叫做科学记数法。5.12e7.

### 问题7：
计算下式，并用科学计数法表示结果：3.14e7*0.8

In [7]:
# Try your code here 
print(3.14e7*8)

251200000.0


### 回答7：（双击在下面回答问题）
2.512e*7

### 问题8：
int（3.14159）和float（3）分别会输出什么结果？为什么？

In [8]:
# Try your code here 
print(int(3.14159))
print(float(3))

3
3.0


### 回答8：（双击在下面回答问题）
3和3.0。int（）是强制转换成整型，float（）是强制转化成浮点型。

### 问题9：
表达式运行后的结果，在未再次运行表达式的情况下，会随着表达式的改变而改变么？为什么？

In [9]:
# Try your code here 
a = 3
b = 4
c = a*b
print(c)
b = 5
print(c)

12
12


### 回答9：（双击在下面回答问题）
不会改变，表达式计算结果赋值给变量后，变量会忘记现在的表达式，更改表达式后，必须再次运行，再能对变量再次赋值。

### 问题10：
4,"4"和4.0表示的含义相同么？为什么？

In [10]:
# Try your code here 
print(type(4))
print(type("4"))
print(type(4.0))

<class 'int'>
<class 'str'>
<class 'float'>


### 回答10：（双击在下面回答问题）
不相同。4表示整型数据，“4”表示字符串数据，4.0是浮点型数据。

### 问题11：
x=4，y='4',      print(x\*4)和print(y\*4)的输出结果相同么？为什么？

In [11]:
# Try your code here 
x = 4
y = '4'
print(x*4)
print(y*4)

16
4444


### 回答11：（双击在下面回答问题）
不相同，x是整型，x\*4得到的也是整型数据12，y是字符串，y\*4得到的依旧是字符串，4444.

### 问题12：
检查数据类型的指令是什么？请举例说明使用方法和输出结果

In [12]:
# Try your code here 
print(type("Udacity"))

<class 'str'>


### 回答12：（双击在下面回答问题）
type()。例如type("Udacity")输出类型为str。

### 问题13：
'='和'=='分别表示什么含义？print(2==5)输出什么结果？属于什么类型数据？

In [13]:
# Try your code here 
print(2==5)
a = 3
print(a)

False
3


### 回答13：（双击在下面回答问题）
'='表示赋值将符号右边的值赋给左边，'=='是判断符号两边是否相等，print（2==5）会输出false，为布尔型。

### 问题14：
逻辑运算符主要有哪些？分别在什么情况下整个表达式结果为True？

In [14]:
# Try your code here 
x = True
y = True
print(x and y)
y = False
print(x or y)
print(not y)

True
True
True


### 回答14：（双击在下面回答问题）
- and	（x and y）	如果x为false，返回false，否则返回y的值
- or	（x or y）	如果x为true，返回true，否则返回y的值
- not	（not x）	如果x为true返回false，如果x为false返回true

### 问题15：
string1='I'm a student.'          print(string1)结果会报错么？如何修改string1使其得到正确的输出？

In [15]:
# Try your code here 
string1 ="I'am a Student."
print(string1)
string2 = 'I\'am a Student.'
print(string2)

I'am a Student.
I'am a Student.


### 回答15：（双击在下面回答问题）
会报错，由于字符串中出现了',而字符串两边也是' '，可以将字符两边改为""，也可以在字符串中的’前加上\.